### PR TITLE
Building _hashlib and _ssl modules

### PR DESCRIPTION
**Pull request by izumiberat**
I am using Mac OS Catalina 10.15.7.

To avoid getting the following messages : 

> The necessary bits to build these optional modules were not found:
> _hashlib              _ssl   

as well as:

> Could not build the ssl module!
> Python requires a OpenSSL 1.1.1 or newer

Based on the comments below, the solution is to indicate the OpenSSL version in the brew command: `brew install openssl@1.1` and make a note to update this line with `brew install openssl@3` when OpenSSL 3 is supported.

Initial (and outdated descritption below):
<details>
Running `brew install openssl@3` allowed the  _hashlib _ssl modules to build, but as suggested in a comment by @ezio-melotti it seemed strange that the version needed to be specified. 

I am therefore updating the documentation by leaving the suggested command line as `brew install openssl` rather than `brew install openssl@3 ` but I am adding a note saying that if user gets the message 

> Could not build the ssl module!
> Python requires a OpenSSL 1.1.1 or newer 

they can try specifying the version of OpenSSL by using `brew install openssl@3`.
</details>

Issue number: https://github.com/python/cpython/issues/93695

### CODE REVIEW

**Comment by tiran** [comment]
Please be aware that OpenSSL 3.0 is still not supported by Python. It's possible to compile Python with OpenSSL 3.0 but it's not tested well. I advise against examples that refer to OpenSSL 3.0.

**Comment by ezio-melotti** [comment]
> Please be aware that OpenSSL 3.0 is still not supported by Python. 

If it's not supported, documenting a command that installs it sounds wrong, so I think it's fine to document `openssl@1` (or `openssl@1.1`).

@tiran: are there plans to add support for OpenSSL 3?  If there is an issue already, a note about updating the devguide could be added there so that we don't forget.

File: setup.rst [hunk 1]
```diff
@@ -401,6 +401,13 @@ and ``make``::
 
     $ make -s -j2
 
+.. note::
```
Comment (ezio-melotti) [inline setup.rst:404]: There's a trailing space here.
I'd recommend to configure your editor to remove them automatically.

File: setup.rst [hunk 2]
```diff
@@ -401,6 +401,13 @@ and ``make``::
 
     $ make -s -j2
 
+.. note:: 
+   If the _hashlib and _ssl modules do not build, specifying the openssl version can resolve the issue. As indicated earlier, you might need to run ``make clean`` before or after re-running ``configure``.
```
Comment (ezio-melotti) [inline setup.rst:405]: This line is too long.

**Review by ezio-melotti — COMMENTED** [review]
I left a couple of comments, but you can wait until decide how to fix this issue before updating the PR.

File: setup.rst (hunk 1 above)
Comment (izumiberat) [reply setup.rst:404]: Thanks very much for the editor advice @ezio-melotti. I was updating the doc directly online on github but I will get to using an editor soon enough :) I will wait to hear back from @tiran as I will certainly remove completely the note and update the brew line command.

File: setup.rst (hunk 2 above)
Comment (izumiberat) [reply setup.rst:405]: Thanks for the feedback. Agreed.

**Comment by izumiberat** [comment]
> If it's not supported, documenting a command that installs it sounds wrong, so I think it's fine to document `openssl@1` (or `openssl@1.1`).

@ezio-melotti the [homebrew documentation](https://formulae.brew.sh/formula/) provides two options when looking up `openssl`: openssl@1.1 or openssl@3. We could therefore go with 1.1 for now.

**Comment by erlend-aasland** [comment]
> @tiran: are there plans to add support for OpenSSL 3? If there is an issue already, a note about updating the devguide could be added there so that we don't forget.

Yes, there are multiple open issues related to OpenSSL 3 in the tracker, so adding a note would be fine.

File: setup.rst (hunk 1 above)
Comment (ezio-melotti) [reply setup.rst]: From GitHub it's more difficult to catch trailing whitespace -- I sometimes do this mistake myself.  I started to check that manually before submitting the PR by highlighting the text.  Even if you still miss one, the checks on the PR will tell you and you can fix the error in another commit.

**Comment by ezio-melotti** [comment]
> Yes, there are multiple open issues related to OpenSSL 3 in the tracker, so adding a note would be fine.

Is there a meta-issue or anyway an issue that once closed will allow the switch to `openssl@3`?  If so, the note should be added there.  Another option is to create a project to group the OpenSSL issues, including a draft issue to update the devguide.

**Comment by erlend-aasland** [comment]
> Is there a meta-issue or anyway an issue that once closed will allow the switch to `openssl@3`? If so, the note should be added there. Another option is to create a project to group the OpenSSL issues, including a draft issue to update the devguide.

I don't think there's a meta issue, so creating a project SGTM.

**Comment by izumiberat** [comment]
@ezio-melotti I am unclear as to where needs to be noted the switch to `openssl@3`, but this is what could be added:

> **Note for when OpenSSL3 becomes supported**
Please update the documentation [Getting Started](https://devguide.python.org/setup/#macos-and-os-x): `brew install openssl@1.1` needs to be updated with `brew install openssl@3`. It is related to https://github.com/python/devguide/pull/892.

Once we made a note of this somewhere (happy to do it if you can tell me where), this PR could be merged as it is.

**Review by ezio-melotti — APPROVED** [review]

**Comment by ezio-melotti** [comment]
I created a project about OpenSSL 3 with a draft issue about updating the devguide:
https://github.com/orgs/python/projects/28/views/1

**Review by erlend-aasland — APPROVED** [review]

**Comment by izumiberat** [comment]
> project about OpenSSL 3 with a draft issue about updating the devguide:
> https://github.com/orgs/python/projects/28/views/1

@ezio-melotti I get a 404 error message. If it is a permission issue, don't worry about it.

**Comment by izumiberat** [comment]
@erlend-aasland or @ezio-melotti does one of you want to merge this PR to main?

**Comment by erlend-aasland** [comment]
Thanks for the heads-up and thanks for the PR!

**Pull request merged**